### PR TITLE
Add Creality Slicer v4.8.2-96

### PR DESCRIPTION
**Pull request by vahrennd**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/creality-slicer.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+cask "creality-slicer" do
+  version "4.8.2-96"
+  sha256 :no_check
```
Comment (vahrennd) [inline Casks/creality-slicer.rb:3]: Just wanted to check if this was desirable here. `brew audit` said to use no sha check if there was no version in the download URL, but there is what looks like a unique ID that may change if the app version changes.

Comment (neutric) [reply Casks/creality-slicer.rb:3]:
```
%  curl -ILs https://file2-cdn.creality.com//website/b049ecf6-2295-4fac-8d10-377dc67a494c | grep -E "(Location|Disposition)"
Content-Disposition: attachment;filename="Creality_Slicer-4.8.2-build-96-Darwin.zip"
```
If the hex string is static across versions, then we need `sha256 :no_check`. If it changes with every new version, than we'd have to carry it along as part of `version` but would gain checksumming. archive.org has no history on that page, so I guess we'd have to wait for an update to ultimately decide.

File: Casks/creality-slicer.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  url "https://file2-cdn.creality.com/website/b049ecf6-2295-4fac-8d10-377dc67a494c"
+  name "creality-slicer"
```
Comment (neutric) [inline Casks/creality-slicer.rb:6]: `name "Creality Slicer"`

File: Casks/creality-slicer.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  livecheck do
+    url :homepage
+    regex(/Creality_Slicer-((?:\d+(?:\.\d+)*)-build-(?:\d+(?:\.\d+)*))-Darwin\.zip/i)
+    strategy :page_match do |page, regex|
+      page.scan(regex).map { |match| match[0].sub("-build-", "-") }
+    end
+  end
+
+  app "Creality Slicer.app"
```
Comment (neutric) [inline Casks/creality-slicer.rb:18]: Please try to add a [`zap` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap). The linked documentation provides some recommended methods to create it.

File: Casks/creality-slicer.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  desc "Slicer for all Creality FDM 3D printers"
+  homepage "https://www.creality.com/download/"
+
+  livecheck do
+    url :homepage
+    regex(/Creality_Slicer-((?:\d+(?:\.\d+)*)-build-(?:\d+(?:\.\d+)*))-Darwin\.zip/i)
+    strategy :page_match do |page, regex|
+      page.scan(regex).map { |match| match[0].sub("-build-", "-") }
+    end
+  end
```
Comment (neutric) [inline Casks/creality-slicer.rb:16]: Not sure much is won by stripping out "-build" from the version. The resulting version scheme looks different to what you find on `homepage` and thus might cause mental overhead without benefit that I can see. Wait for a maintainer's comment on that, though.

**Review by neutric — COMMENTED** [review]
Thank you for your contribution to `homebrew-cask` @vahrennd! My review is in the code. Take it with a grain of salt as I am not a maintainer.

File: Casks/creality-slicer.rb (hunk 4 above)
Comment (vahrennd) [reply Casks/creality-slicer.rb:16]: The reason I did so is that everywhere I can find references version 4.8.2-96 except the website. For example, in the app itself:

image

That being said, if the consensus is to just leave the `-build-` part in then I can go with that.

File: Casks/creality-slicer.rb (hunk 1 above)
Comment (vahrennd) [reply Casks/creality-slicer.rb:3]: Maybe we can just leave it without the check for now and see what happens next time they release a new version.

File: Casks/creality-slicer.rb (hunk 4 above)
Comment (neutric) [reply Casks/creality-slicer.rb:16]: In that case 4.8.2-96 is to be preferred.

File: Casks/creality-slicer.rb [hunk 5]
```diff
@@ -9,18 +9,18 @@
 
   livecheck do
     url :homepage
-    regex(/Creality_Slicer-((?:\d+(?:\.\d+)*)-build-(?:\d+(?:\.\d+)*))-Darwin\.zip/i)
+    regex(/Creality[._-]Slicer[._-](\d+(?:\.\d+)+)[._-]build[._-](\d+)[._-]Darwin\.zip/i)
     strategy :page_match do |page, regex|
-      page.scan(regex).map { |match| match[0].sub("-build-", "-") }
+      page.scan(regex).map { |match| "#{match[1]},#{match[2]}" }
```
Comment (neutric) [inline Casks/creality-slicer.rb]: `page.scan(regex).map { |match| "#{match[0]},#{match[1]}" }`

**Comment by miccal** [comment]
@vahrennd I am assuming that you need access to a Creality FDM 3D printer for this softeware to be useful?

If so, this needs to be submitted to the [`homebrew-cask-drivers` repository](https://github.com/Homebrew/homebrew-cask-drivers).

**Comment by vahrennd** [comment]
You would need access to *a* 3D printer to make much use of it, sure. It technically has preset profiles for other brands of printers as well as the ability to define your own. [ultimaker-cura](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/ultimaker-cura.rb) is basically an equivalent piece of software available in homebrew-cask which is what made me think it belonged here.

Would it help if I updated the description to more accurately reflect that? I mostly just took that from their download page.

**Comment by miccal** [comment]
> You would need access to _a_ 3D printer to make much use of it, sure. It technically has preset profiles for other brands of printers as well as the ability to define your own. [ultimaker-cura](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/ultimaker-cura.rb) is basically an equivalent piece of software available in homebrew-cask which is what made me think it belonged here.
> 
> Would it help if I updated the description to more accurately reflect that? I mostly just took that from their download page.

Thanks for the explanation -- I think actually it will be fine in this repository.

**Pull request merged**